aws/terraform/infra/vpc.tf

```diff
@@ -0,0 +1,159 @@
+resource "aws_vpc" "this" {
+  cidr_block           = "10.0.0.0/16"
+  enable_dns_hostnames = true
+
+  tags = merge(
+    module.common_vars.tags,
+    {
+      Name = "${local.resource_prefix}-vpc-${local.resource_suffix}"
+    }
+  )
+}
+
+/*
+ Setup a gateway between aws VPC and internet. Allow access to and from Resources
+ in subnet with public IP addr.
+ Ref: https://nickcharlton.net/posts/terraform-aws-vpc.html
+*/
+resource "aws_internet_gateway" "this" {
+  vpc_id = aws_vpc.this.id
+
+  tags = merge(
+    module.common_vars.tags,
+    {
+      Name = "${local.resource_prefix}-internet-gateway-${local.resource_suffix}"
+    }
+  )
+}
+
+/*
+ Grants us ability to yield different availability zones for a region
+*/
+data "aws_availability_zones" "available" {
+  state = "available"
+}
+
+/*
+ Create a public Subnet that the two follow private Subnets can use for internet egress connections.
+ Contains our NAT Gateway.
+*/
+resource "aws_subnet" "public" {
+  vpc_id                  = aws_vpc.this.id
+  cidr_block              = "10.0.32.0/20"
```

Comment:
JFYI - These `cidr_blocks` are currently configurable in the CFN template.

Reply:
Oh really? That's really good to know. I think that making this configurable would make for another good Github issue and separate PR to get closer to the existing CFN template